### PR TITLE
Update the advantage dashboard to display ESM apps contracts

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Add ESM apps to be displayed in contracts dashboard on the advantage page
- Linted the styles.scss file

**Notes for UX**
There is nothing clear in the response that indicates weither the contract is exclusively an apps or infra contract other than the esm affordance. I wonder if it's clear enough with the "Apps" and "Infra" text in the esm cell?

## QA
- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8001/advantage
- If you have contracts check it matches [the wireframe](https://user-images.githubusercontent.com/19801137/76755183-53a23880-677b-11ea-9053-b1d33971c295.png) if not check the screenshots

## Issue / Card
Fixes https://github.com/canonical-web-and-design/web-squad/issues/2353

## Screenshots
Screenshot_2020-03-19 Ubuntu Advantage for Infrastructure Ubuntu

Screenshot_2020-03-19 Ubuntu Advantage for Infrastructure Ubuntu(1)

Screenshot_2020-03-19 Ubuntu Advantage for Infrastructure Ubuntu(2)

### CODE REVIEW

**Comment by matthewpaulthomas** [comment]
The screenshot looks fine. Unfortunately I can’t try out the UI more than that, because there’s no mock paid sub.

> There is nothing clear in the response that indicates weither the contract is exclusively an apps or infra contract other than the esm affordance. I wonder if it's clear enough with the "Apps" and "Infra" text in the esm cell?

I think it’s fine like that as long as there aren’t any Apps subscriptions. The moment there are, it should be fixed at the response level.

**Comment by anthonydillon** [comment]
Setting to UX +1 as the comment above seems happy.

**Comment by anthonydillon** [comment]
@lechsandecki can you QA it and let me know your happy and ill merge it.

**Comment by anthonydillon** [comment]
Addressed all the actionable QA points.

File: webapp/views.py [hunk 1]
```diff
@@ -178,22 +183,29 @@ def advantage_view():
...
-                            contract["contractInfo"]["effectiveFrom"],
-                            "%Y-%m-%dT%H:%M:%S.%fZ",
-                        ).strftime(
-                            "%d %B %Y"
+
+                    contract["contractInfo"]["status"] = "active"
+                    if contract["contractInfo"].get("effectiveTo"):
+                        effectiveTo = datetime.strptime(
+                            contract["contractInfo"]["effectiveTo"],
+                            "%Y-%m-%dT%H:%M:%SZ",
```
Comment (nottrobin) [inline webapp/views.py:197]: Thought you were gonna use dateutil?

image

**Comment by anthonydillon** [comment]
I didn't get a chance to work out the package by the end of the date. I can give it a go at some point tomorrow.

File: webapp/views.py [hunk 2]
```diff
@@ -178,22 +183,29 @@ def advantage_view():
...
-                            "effectiveFromFormatted"
-                        ] = datetime.datetime.strptime(
-                            contract["contractInfo"]["effectiveFrom"],
-                            "%Y-%m-%dT%H:%M:%S.%fZ",
-                        ).strftime(
-                            "%d %B %Y"
+
+                    contract["contractInfo"]["status"] = "active"
+                    if contract["contractInfo"].get("effectiveTo"):
+                        effectiveTo = datetime.strptime(
```
Comment (nottrobin) [inline webapp/views.py:195]: This is not how we write variable names in Python - see https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles. I'll fix this and commit it if I can.

**Comment by nottrobin** [comment]
I updated your branch some, fixed the variable name and the timezone issue.

Now I see:

image

Looks a bit wrong - too much space below "Effective to", and I think the line in between the two looks a bit weird. Also, didn't you say there'd be a "status" property displayed?

**Comment by anthonydillon** [comment]
Review document: https://docs.google.com/document/d/1ok690GOC-jOUc9aP9XHq5Sm7vzZ0yXPtDZdI_C5pDsM/edit#

**Comment by anthonydillon** [comment]
@nottrobin as we are limited on designers do you think we can land this and update once the design is available.

**Comment by sowasred2012** [comment]
@anthonydillon @nottrobin re: design, I can pick that up in a renewals PR, as I'm going to be doing some anyway - lyubo has a mockup [here](https://app.zeplin.io/project/5e70c2e3d8982a16dfc6ce94/screen/5e821403148f9315628f090c) that touches this.

**Comment by anthonydillon** [comment]
@sowasred2012 thanks. Unfortately all reviewers are off now so will have to wait till tomorrow to get a QA+1.

**Pull request merged**